lib/active_merchant/billing/gateways/conekta.rb

```diff
@@ -173,22 +173,27 @@ def parse(body)
         JSON.parse(body)
       end
 
-      def headers(meta)
+      def headers(options)
         {
-          "Accept" => "application/vnd.conekta-v#{options[:version]}+json",
+          "Accept" => "application/vnd.conekta-v#{@options[:version]}+json",
           "Accept-Language" => "es",
-          "Authorization" => "Basic " + Base64.encode64("#{options[:key]}:"),
+          "Authorization" => "Basic " + Base64.encode64("#{@options[:key]}:"),
           "RaiseHtmlError" => "false",
           "Conekta-Client-User-Agent" => {"agent"=>"Conekta ActiveMerchantBindings/#{ActiveMerchant::VERSION}"}.to_json,
-          "X-Conekta-Client-User-Agent" => user_agent,
-          "X-Conekta-Client-User-Metadata" => meta.to_json
+          "X-Conekta-Client-User-Agent" => conekta_client_user_agent(options),
+          "X-Conekta-Client-User-Metadata" => @options.to_json
```

Comment:
Shouldn't this be either `options[:meta].to_json` or `@options[:meta].to_json`? If not, then I really _am_ getting confused what's going on here.

Reply:
I was also thrown for a loop by this and this was the result of trial and error, so your confusion is a good signal that I should take a step back and really learn what's going on. `@options[:meta]` doesn't find anything when used here. Previously, the `headers` method pulled it from `options[:meta]`. I'll root around a bit more.

Reply:
I think I know what's going on here, though it's kind of a doozy to explain. Here goes. When the gateway was first added, `commit` called `headers(options[:meta])`, and `headers` had `meta.to_json` for the appropriate header, anticipating a `meta` option field being passed to it. However, none of the calls to `commit` passed `options`. So that has sat there doing nothing since then. But `headers` also called `options[:key]` and `options[:version]`, which, in the absence of a local `options`, apparently resolves to the global values? I'm still not sure why that happens, but I've confirmed in `pry`. 

In any case, my confusion led me to think the Metadata header actually wanted the global options hash, which.... no. So I'll restore it to look for `options[:meta]`.

Reply:
@bpollack Done, and added test for `options[:meta]`

Reply:
> But headers also called options[:key] and options[:version], which, in the absence of a local options, apparently resolves to the global values?

Yep, because [`@options` is declared `attr_reader` in `gateway.rb`](https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateway.rb#L146), so if there's no local variable called `options`, then you'll grab the gateway's global config from the accessor.

In related news we may want to stop naming variables `options` in gateways…